New guidelines for clone location (#20)
worldbank/dime-github-trainings: GitHub-resources/DIME-GitHub-Guides/clone-location.md

on why synced folders are not good locations for clones.

In order to not have to remember what locations are good locations
each time you clone a repository, it is a good practice to create a folder
called `git`, `github`, `github-clones` or something similar
where _all_ clones for _all_ projects you work on are stored.
This way you only have to find a good location for your clones once,
create this folder there and then clone all your clones in that folder.

A common location for your clone is `C:\Users\username\github` on Windows computers
(more specifically `C:\Users\wb123456\github` on WB computers
where `123456` is replaced with your UPI)
or `/Users/username/github` on Mac computers.
Another common location is `C:\Users\username\Documents\github`
or `/Users/username/Documents/github`, however,
this location is acceptable only if OneDrive is not installed on your computer
as then the `\Documents\` folder is also synced.
OneDrive is commonly installed on Windows 10 computers,
and is always installed on all WB computers.

The best way to know if your folder is synced in OneDrive
is to check if the folder has the _Status_ column
when viewed in the file explorer.

> Any folder with this column (see image blow) is indeed synced in OneDrive.

Reviewer: ```suggestion
Any folder with this column (see image below) is indeed synced in OneDrive 
and **should not** be used to store clones.
```

then you should delete the folder called `lyrics-clone` and all its content.

If you open GitHub Desktop and select this repository as your current repository,
then you will see the screen shown in the image below.
Then simply click the button "_Clone Again_".
If you do not see this screen or
if you are not able to select the repository as current repository,
then just go to GitHub.com and clone it again.

<img src="https://github.com/worldbank/dime-github-trainings/blob/move-clone/GitHub-resources/DIME-GitHub-Guides/img/clone-not-found.png" width="75%"><!--- Image is read from master branch or use full URL-->

This method can also be used if you have some issues with a clone
you know that you have no un-committed/pushed and no ignored files and
and want to start over fresh.
And in this case you can of course clone to the same location.

#### Method 2: Move the existing clone

If you want to move the clone and keep all content,
including content not pushed to GitHub.com
such as un-committed edits and ignored files,
then you need to use this method.

###### M2: How to move the clone

First start by copying the clone folder to the new location.

> Remember that you need to copy the folder as well as its content.

Reviewer: I don't quite understand this remark. Are you clarifying that you need to copy the folder and not only the files and subfolders of the folder? If that's the case, I think it's enough to say that you need to copy the folder, because you couldn't possibly move the folder without moving its content as well

Author: ```suggestion
Remember that you need to copy the folder and not just its content.
```
Haha, that is true. What I meant was again to repeat that both folder and content needs to be copied. I made it sound as if the risk was that the folder would be copied without the content, while the risk is the other way around. Let me know if this sounds better!

Reviewer: yes, this is better!

and there is no standard workflow where this should ever be the norm.

#### Non-shared synced folders

What about non-shared synced folders? Then there will never be an issue with another user.
While that is true, there are still errors than can happen in any kind of synced folder.
And while these errors are more rare they are
likely to cause major disruptions when they happen.

When syncing software detects a file with edits to be synced to the cloud,
it blocks access to that file for a fraction of a second.
That is fine when it is only much slower humans that also competes for access to those files.
However, git is not a human and in certain git actions,
git can make many updates in a fraction of a second
to the technical files in the `.git` folder.
There is always a `.git` folder in all clone folders,
but depending on the settings of your computer,
this `.git` folder might or might not be visible to you.

As you use different features of GitHub Desktop,
series of git commands are executed in the background in rapid succession.
This means that git might do an edit to a file,
then do something else and then need to make a second edit to that first file.

> This can happen within the same fraction of a second under which  

Reviewer: ```suggestion
This can happen within the same fraction of a second under which
```
Deleting two blank spaces, in markdown they are interpreted as a line break

a syncing software has blocked of access to the file.
Sometimes this is fine, other times GitHub Desktop tells you there is an error,
but sometimes there is an error in these files that is not detected immediately.

This can lead to things like git not being aware that you have changed branch
and commits you think you are committing to one branch,

> is actually pushed to a different branch.

Reviewer: ```suggestion
are actually pushed to a different branch.
```
